app/api/files/storage.ts

```diff
@@ -121,6 +136,37 @@ export const storage = {
     return true;
   },
 
+  async listFiles() {
+    if (tenants.current().featureFlags?.s3Storage) {
+      const results = await s3().list(tenants.current().name);
+      return results.map(c => fromS3Key(c.Key!));
+    }
+
+    const files: { type: FileTypes; filename: string; path: string }[] = [];
+    await Object.keys(paths).reduce(async (prev, type) => {
+      await prev;
+      const filetype = type as keyof typeof paths;
+      try {
+        (await readdir(paths[filetype](''), { withFileTypes: true })).forEach(file => {
+          if (file.isFile()) {
+            files.push({
+              type: filetype,
+              filename: file.name,
+              path: join(paths[filetype](''), file.name),
+            });
+          }
+        });
+      } catch (err) {
+        if (err?.code === 'ENOENT') {
```

Comment:
> What can cause an ENOENT that we ignore it here? The files are already given by a readdir, so those should be there. A missing folder from the ones configured in `path`?

Correct.

> If it's a missing folder, is that ok, since we're auto creating it when they are needed and it's fine that they are not present for a given tenant?

There will be other places in the app that will handle such scenarios. For this particular API, the semantic is "list every file there". I think that, when there's no directory for one type of file, we should assume "no files there" rather than erroring, but it's discussable.